### ISSUE TITLE
Missing response header in OpenAPI documentation

### ISSUE THREAD

**Issue by drcpu-github** [marshmallow-code/flask-smorest#578]
When generating OpenAPI documentation, additional headers in the response decorator are overwritten whenever the paginate decorator is used. This appears to be caused by the `_document_pagination_metadata` in `pagination.py` which overwrites the `resp_doc["headers"]` structure.

A simple fix could be something like this:
```py
def _document_pagination_metadata(self, spec, resp_doc):
    """Document pagination metadata header

    Override this to document custom pagination metadata
    """
    if "headers" not in resp_doc:
        resp_doc["headers"] = {}
    resp_doc["headers"].update({
        self.PAGINATION_HEADER_NAME: "PAGINATION"
        if spec.openapi_version.major >= 3
        else PAGINATION_HEADER
    })
```

**Comment by lafrech** [comment]
Yes, I guess you're right.

Would you like to submit a PR with this fix and a test?

**Comment by drcpu-github** [comment]
Sure thing, I created one.

**Comment by lafrech** [comment]
Fixed in #579.
